reevo.py

```diff
@@ -16,10 +16,23 @@ def __init__(
         root_dir: str, 
         generator_llm: BaseClient, 
         reflector_llm: Optional[BaseClient] = None,
+        
+        # Support setting different LLMs for each of the four operators: 
+        # Short-term Reflection, Long-term Reflection, Crossover, Mutation
+        short_reflector_llm: Optional[BaseClient] = None,
+        long_reflector_llm: Optional[BaseClient] = None,
+        crossover_llm: Optional[BaseClient] = None,
+        mutation_llm: Optional[BaseClient] = None
     ) -> None:
         self.cfg = cfg
         self.generator_llm = generator_llm
         self.reflector_llm = reflector_llm or generator_llm
+
+        self.short_reflector_llm = short_reflector_llm
+        self.long_reflector_llm = long_reflector_llm
+        self.crossover_llm = crossover_llm
+        self.mutataion_llm = mutation_llm
```

Comment:
I think it's more elegant to implement it like this 🤔:
```python
self.short_reflector_llm = short_reflector_llm or self.reflector_llm
self.long_reflector_llm = long_reflector_llm or self.reflector_llm
self.crossover_llm = crossover_llm or generator_llm
self.mutation_llm = mutation_llm or generator_llm
```
So that we don't have to write sooooo many conditional expressions in L376, L395, L434 and L456-459.
BTW, it should be `self.mutation_llm` instead of `self.mutataion_llm`.